BattleNetwork/bnMessage.cpp

```diff
@@ -15,7 +16,8 @@ void Message::OnUpdate(double elapsed) {
 void Message::OnDraw(sf::RenderTarget& target, sf::RenderStates states) {
   auto bounce = std::sin((float)totalElapsed * 20.0f);
 
-  nextCursor.setPosition(sf::Vector2f(GetTextBox()->GetFrameWidth() - 15.0f, 5.0f + bounce));
+  auto textBoxBottom = (GetTextBox()->getPosition().y + GetTextBox()->GetFrameHeight() / 2.0f) / 2.0f;
```

Comment:
use renderstates .getTransform() to match scale. Someone might want a 1:1 resolution of their game, instead of the 2x upscale in ours, and the textbox and its items should still be in the right place.

Reply:
I think currently all of the menus I checked have hardcoded values to fit 2x scaling or have a hardcoded *2.0f somewhere. Not sure where the scale is set, thinking about going back and testing everything with 1x scale if that's the case.

Reply:
I understand the confusing. the finished widgets and nodes are upscaled in the scene but if it's an internal node it should also use the parent's transforms. this way when/if we upscale or move the parent (textbox) everything is in the right spot

Reply:
bnTextBox.cpp and bnAnimatedTextBox.cpp currently set scaling internally as well. Message based classes don't seem to have a way to set scale. I'm not too sure on how to go about fixing that, so mostly leaving these as stuff to look at.